### PR TITLE
Print deprecate message once, if console is defined

### PR DESCRIPTION
**Pull request by deian**
I'm using the library (thanks for this shim BTW) in an environment where console is not yet defined. Mind adding a patch along these lines to accommodate for this? The patch does this, and like node.j's util.deprecate, it only prints the message once.

### CODE REVIEW

**Comment by scotthovestadt** [comment]
+1

**Comment by deian** [comment]
@tvcutsem ping :)

File: reflect.js [hunk 1]
```diff
@@ -540,6 +540,25 @@ function testIntegrityLevel(target, level) {
...
+ * environments where global.console is defined.
+ *
+ * @param message the deprecation warning message.
+ */
+function deprecate(deprecated, message) {
+ var warned = false;
+ var warn = global.console ? console.warn : function () { };
+ return function () {
+    warned = true;
+    warn(message);
```
Comment (tvcutsem) [inline reflect.js]: shoudn't `warn` be called conditionally only if `!warned`?

File: reflect.js [hunk 2]
```diff
@@ -658,6 +678,7 @@ Validator.prototype = {
       throw new TypeError("cannot report a non-configurable descriptor "+
                           "for configurable or non-existent property '"+name+"'");
     }
+    */
```
Comment (tvcutsem) [inline reflect.js]: Can you comment on why this invariant check must be removed? Also, since this change has nothing to do with this PR's original request (printing deprecation warning only once) it should probably go in its own PR?

**Comment by tvcutsem** [comment]
@deian Sorry for the long wait. Yes, this looks reasonable. See my in-line comments before accepting.

**Comment by deian** [comment]
Thanks! Will fix! (Sorry the second commit has nothing to do with the original PR. Github fail :()

**Comment by tvcutsem** [comment]
I decided to simply remove the deprecation notice. See issue #66 for the rationale. That makes this PR obsolete.

Thanks BTW for the reference to `util.deprecate`. I didn't yet know about it and it seems like the standard way to flag deprecated methods on node. One more reason to simply remove the `console.warn` or `util.deprecate` is that it avoids dependencies on node.js, since `reflect.js` should otherwise also load fine in other environments.

**Comment by deian** [comment]
Fix looks good to me, thanks!
